eth/catalyst/simulated_beacon.go

```diff
@@ -144,6 +144,16 @@ func (c *SimulatedBeacon) sealBlock(withdrawals []*types.Withdrawal) error {
 	feeRecipient := c.feeRecipient
 	c.feeRecipientLock.Unlock()
 
+	// Reset to CurrentBlock in case of the chain was rewound
+	block := c.eth.BlockChain().CurrentBlock()
+	if c.curForkchoiceState.HeadBlockHash != block.Hash() {
+		c.curForkchoiceState = engine.ForkchoiceStateV1{
+			HeadBlockHash:      block.Hash(),
+			SafeBlockHash:      block.Hash(),
+			FinalizedBlockHash: block.Hash(),
```

Comment:
We should set the finalized block hash similarly as we do below, and put the shared logic into a `getFinalizedHashByNumber` function.

Reply:
thanks, please take another look.

Reply:
Ah, sorry if I was too vague.  I meant that we should set the finalized block hash to that of the last epoch block when we have to reset the `curForkChoiceState` because of a chain rewind.